ambry-api/src/test/java/com.github.ambry/account/AccountContainerTest.java

```diff
@@ -0,0 +1,574 @@
+  // Reference Account fields
+  private final short refAccountId = 1234;
+  private final String refAccountName = "testAccount1";
+  private final AccountStatus refAccountStatus = AccountStatus.ACTIVE;
+  private final JSONObject accountJsonLike = new JSONObject();
+  // Reference Container fields
+  private final List<Short> refContainerIds = Arrays.asList(new Short[]{0, 1});
+  private final List<String> refContainerNames = Arrays.asList(new String[]{"Container_0_0", "Container_0_1"});
+  private final List<String> refContainerDescriptions =
+      Arrays.asList(new String[]{"Public container", "Private container"});
+  private final List<ContainerStatus> refContainerStatuses =
+      Arrays.asList(new Container.ContainerStatus[]{ContainerStatus.ACTIVE, ContainerStatus.INACTIVE});
+  private final List<Boolean> refContainerPrivacies = Arrays.asList(new Boolean[]{false, true});
+  private final List<JSONObject> containerJsonLikeList = new ArrayList<>();
+
+  /**
+   * Initialize the metadata in JsonObject for account and container.
+   * @throws JSONException
+   */
+  @Before
+  public void init() throws JSONException {
+    accountJsonLike.put(JSON_VERSION_KEY, JSON_VERSION_1);
+    accountJsonLike.put(ACCOUNT_ID_KEY, refAccountId);
+    accountJsonLike.put(ACCOUNT_NAME_KEY, refAccountName);
+    accountJsonLike.put(STATUS_KEY, refAccountStatus);
+    accountJsonLike.put(CONTAINERS_KEY, new JSONArray());
+    JSONArray containerArray = new JSONArray();
+    for (int i = 0; i < 2; i++) {
+      JSONObject containerJsonLike = new JSONObject();
+      containerJsonLike.put(Container.JSON_VERSION_KEY, Container.JSON_VERSION_1);
+      containerJsonLike.put(CONTAINER_ID_KEY, refContainerIds.get(i));
+      containerJsonLike.put(CONTAINER_NAME_KEY, refContainerNames.get(i));
+      containerJsonLike.put(Container.STATUS_KEY, refContainerStatuses.get(i));
+      containerJsonLike.put(DESCRIPTION_KEY, refContainerDescriptions.get(i));
+      containerJsonLike.put(IS_PRIVATE_KEY, refContainerPrivacies.get(i));
+      containerJsonLike.put(PARENT_ACCOUNT_ID_KEY, refAccountId);
+      containerJsonLikeList.add(containerJsonLike);
+      containerArray.put(containerJsonLike);
+    }
+    accountJsonLike.put(CONTAINERS_KEY, containerArray);
+  }
+
+  /**
+   * Tests constructing an {@link Account} from Json metadata.
+   * @throws Exception Any unexpected exceptions.
+   */
+  @Test
+  public void testConstructAccountFromJson() throws Exception {
+    assertAccountAgainstReference(Account.fromJson(accountJsonLike), true, true);
+  }
+
+  /**
+   * Tests constructing {@link Account} and {@link Container} using individual arguments.
+   */
+  @Test
+  public void testConstructAccountAndContainerFromArguments() throws JSONException {
+    List<Container> containers = new ArrayList<>();
+    for (int i = 0; i < 2; i++) {
+      containers.add(new Container(refContainerIds.get(i), refContainerNames.get(i), refContainerStatuses.get(i),
```

Comment:
shouldn't we use Builder here. coz, no one outside the package will have access to constructor.